### PR TITLE
Add the FullNameField component

### PR DESCRIPTION
**Pull request by patrickpeinanw**
*Issue:*https://github.com/bcgov/entity/issues/17974

*Description of changes:*
Add the FullNameField component, its unit test case, and the cypress e2e test.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the namex license (Apache 2.0).

### CODE REVIEW

File: btr-web/btr-common-components/cypress/e2e/pages/examples/form.cy.ts [hunk 1]
```diff
@@ -29,4 +29,79 @@ describe('forms -> validate that form component work inside form', () => {
...
+
+    const name1 = 'François';
+    const name2 = 'José 玛丽';
+    
+    cy.get('#testFullName').type(name1);
+    cy.get('#exampleSubmitButton').click();
+    cy.get('#testFullName').clear().type(name2).blur();
+    cy.get('#exampleSubmitButton').click();
+  });
 })
```
Comment (hfekete) [inline btr-web/btr-common-components/cypress/e2e/pages/examples/form.cy.ts:107]: Nice !

File: btr-web/btr-common-components/pages/examples/form.vue [hunk 2]
```diff
@@ -23,15 +24,26 @@
...
 const schema = z.object({
-  email: z.string().email(t('errors.validation.invalidEmail'))
+  email: z.string().email(t('errors.validation.invalidEmail')),
+  fullName: z.preprocess(normalizeName, 
+    z.string()
+      .min(minNameLength, t('errors.validation.emptyName'))
+      .max(maxNameLength, t('errors.validation.maxNameLengthExceeded'))
+      .refine(validateNameCharacters, t('errors.validation.specialCharacter'))
+  )
 })
```
Comment (hfekete) [inline btr-web/btr-common-components/pages/examples/form.vue:41]: Good job ironing this out!

**Review by hfekete — APPROVED** [review]

**Review by hfekete — CHANGES_REQUESTED** [review]
You will need to pull and rebase and repush. 
There are some changes in the main extra that cannot be resolved in squash and merge.

**Review by kialj876 — COMMENTED** [review]
Looks good to me, but needs the rebase like Hrvoje mentioned

**Review by hfekete — APPROVED** [review]

**Pull request merged**